vision_evaluation/retrieval_evaluators.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import sklearn.metrics as sm
+
+from .evaluators import MemorizingEverythingEvaluator, PrecisionRecallCurveMixin
+from vision_evaluation.prediction_filters import TopKPredictionFilter
+
+
+class RetrievalEvaluator(MemorizingEverythingEvaluator):
+    def add_predictions(self, predictions, targets):
+        """ Add a batch of predictions for evaluation.
+        Args:
+            predictions: the model output array. Shape (N, M) where N is the number of queries and M is the number of items in the dataset
```

Comment:
Can we be more specific about the value type here?

For example, the groundtruth is bool or match level integer, etc?